src/blockchain/rpc.rs

```diff
@@ -441,6 +441,59 @@ fn list_wallet_dir(client: &Client) -> Result<Vec<String>, Error> {
     Ok(result.wallets.into_iter().map(|n| n.name).collect())
 }
 
+/// Factory of [`RpcBlockchain`] instances, implements [`BlockchainFactory`]
+///
+/// Internally caches the node url and authentication params and allows getting many different [`RpcBlockchain`]
+/// objects for different wallet names and with different rescan heights.
+///
+/// ## Example
+///
+/// ```no_run
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let factory = RpcBlockchainFactory {
+///     url: "http://127.0.0.1:18332",
+///     auth: Auth::Cookie {
+///         file: "/home/user/.bitcoin/.cookie".to_string(),
```

Comment:
Needs a PathBuf so `.into()` works.
```suggestion
///         file: "/home/user/.bitcoin/.cookie".into(),
```